util/src/main/java/tc/oc/pgm/util/event/GeneralizedEvent.java

```diff
@@ -48,14 +62,74 @@ public void setCancelled(final boolean cancel) {
     }
   }
 
-  // TODO: Un-implement Physical and EntityAction, since they are SportPaper only
-  @Override
-  public World getWorld() {
-    return getCause() instanceof Physical ? ((Physical) getCause()).getWorld() : null;
+  public @Nullable World getWorld() throws EventException {
+    if (getCause() == null) return null;
+
+    World world = getWorldIfPresent(getCause());
+    if (world != null) return world;
+
+    throw new EventException(getCause().getEventName() + " has no associated world");
   }
 
-  @Override
-  public Entity getActor() {
-    return getCause() instanceof EntityAction ? ((EntityAction) getCause()).getActor() : null;
+  public @Nullable Entity getActor() {
+    return getActorIfPresent(getCause());
+  }
+
+  public static @Nullable Entity getActorIfPresent(Event event) {
```

Comment:
Remove the extra spaces here.

Reply:
Isn't `getActor` is a SportPaper method, right? You would want to use other getters.

Reply:
> Isn't `getActor` is a SportPaper method, right? You would want to use other getters.

Ah, i forgot that.